test/lib/stack.spec.js

```diff
@@ -22,66 +20,76 @@ describe("stack proxy", () => {
     jest.restoreAllMocks()
   })
 
-  describe("when initialized with an cozyURL + token", () => {
-    
+  describe('when initialized with an cozyURL + token', () => {
     const params = { cozyURL, token, onCreateApp, onDeleteApp }
-    
+
     beforeAll(() => {
       jest.clearAllMocks()
-      
+
       stack.init(params)
       stack.get.app()
     })
 
-    it("should call the internal stack init", () => {
-      expect( internal.init ).toHaveBeenCalled()
+    it('should call the internal stack init', () => {
+      expect(internal.init).toHaveBeenCalled()
     })
 
-    it("should not call the cozy-client stack init", () => {
-      expect( client.init ).not.toHaveBeenCalled()
+    it('should not call the cozy-client stack init', () => {
+      expect(client.init).not.toHaveBeenCalled()
     })
 
-    it("should forward requests to the internal stack client", () => {
-      expect( internal.get.app ).toHaveBeenCalled()
+    it('should forward requests to the internal stack client', () => {
+      expect(internal.get.app).toHaveBeenCalled()
     })
 
-    it("should not forward requests to the cozy-client stack client", () => {
-      expect( client.get.app ).not.toHaveBeenCalled()
+    it('should not forward requests to the cozy-client stack client', () => {
+      expect(client.get.app).not.toHaveBeenCalled()
     })
-
   })
 
-  describe("when initialized with a cozy-client instance", () => {
-
-    const cozyClient = { 
+  describe('when initialized with a cozy-client instance', () => {
+    const cozyClient = {
       getStackClient: () => {
         return {
-          token: { token: "mytoken"},
-          uri: "https://test.mycozy.cloud"
+          token: { token: 'mytoken' },
+          uri: 'https://test.mycozy.cloud'
         }
       }
     }
     const params = {
       cozyClient,
       onCreateApp: function() {},
-      onDeleteApp: function() {},
+      onDeleteApp: function() {}
     }
 
     beforeAll(() => {
       jest.clearAllMocks()
-      
+
       stack.init(params)
       stack.get.app()
     })
 
-    it("should call the client stack init", () => {
-      expect( client.init ).toHaveBeenCalled()
+    it('should call the client stack init', () => {
+      expect(client.init).toHaveBeenCalled()
     })
 
-    it("should forward requests to the cozy-client stack client", () => {
-      expect( client.get.app ).toHaveBeenCalled()
+    it('should forward requests to the cozy-client stack client', () => {
+      expect(client.get.app).toHaveBeenCalled()
     })
-    
   })
 
+  it('should throw error if trying to use the stack before init', () => {
+    jest.clearAllMocks()
+    const params = { cozyURL, token, onCreateApp, onDeleteApp }
+
+    // TODO for now the stack is default to internal even before init
+    // expect(() => {
+    //   stack.getStack()
+    // }).toThrowErrorMatchingSnapshot()
+    stack.init(params)
+    expect(() => {
```

Comment:
You probably don't need the check for exceptions as long as you check for marching Snapshots (if there is an exception, your snapshot won't match anyways)

Reply:
It could still throw an exception after the init for some reasons, so I check that after the init, using the stack won't throw any exception
Or maybe I didn't understand your point here

Reply:
If `stack.getStack()` is throwing an exception in L91, it will also do it in  any code is throwing an exception, it will also do it in L93 so the test will fail anyways in L93.

I don't see any case where the L90-92 is adding any protection not found elsewhere. Am I wrong ?

Reply:
The difference is that expecting the call to not throw an execption will make the tests reject, but not fail.

The result is almost the same, without `expect().not.toThrow()` test will "explode" with a dirty Error.
With `expect().not.toThrow()`, the Error will be cleanly caught and the test will reject but with a pretty and nice message.

Reply:
👌